src/pg/brc20/brc20-pg-store.ts

```diff
@@ -110,23 +110,52 @@ export class Brc20PgStore {
     };
   }
 
-  async getHistory(args: { ticker: string } & DbInscriptionIndexPaging): Promise<void> {
-    const results = await this.sql`
-      WITH events AS (
-        SELECT ${this.sql(BRC20_EVENTS_COLUMNS)}
-        FROM brc20_events AS e
-        INNER JOIN brc20_deploys AS d ON d.id = e.brc20_deploy_id
-        INNER JOIN inscriptions AS i ON i.id = e.inscription_id
-        WHERE d.ticker_lower = LOWER(${args.ticker})
-        ORDER BY i.number DESC
-        LIMIT ${args.limit}
-        OFFSET ${args.offset}
-      )
-      SELECT *
-      FROM events
-      INNER JOIN
-    `;
-    // todo: use event history
+  async getActivity(
+    args: {
+      ticker?: string[];
+      block_height?: number;
+      operation?: string[];
+    } & DbInscriptionIndexPaging
+  ): Promise<DbPaginatedResult<DbBrc20Activity>> {
+    const tickerConditions = this.sqlOr(
+      args.ticker?.map(t => this.sql`d.ticker_lower = LOWER(${t})`)
+    );
+
+    const results = await this.sql<(DbBrc20Activity & { total: number })[]>`
+      SELECT
+        ticker,
+        operation,
+        genesis_id AS inscription_id,
+        l.block_height,
+        block_hash,
+        l.tx_id,
+        l.address,
+        max AS deploy_max,
+        d.limit AS deploy_limit,
+        decimals AS deploy_decimals,
+        m.amount AS mint_amount,
+        from_address AS transfer_from,
+        to_address AS transfer_to,
+        t.amount AS transfer_amount,
+        timestamp,
+        COUNT(*) OVER() as total
+      FROM brc20_events AS e
+      INNER JOIN brc20_deploys AS d ON e.brc20_deploy_id = d.id
+      LEFT JOIN brc20_mints AS m ON e.mint_id = m.id
+      LEFT JOIN brc20_transfers AS t ON e.transfer_id = t.id
+      INNER JOIN locations AS l ON e.genesis_location_id = l.id
```

Comment:
I suspect these joins could create weird duplicate row issues unless you add a DISTINCT by inscription_id. You should try that. However, perhaps a better strategy would be to only select from `brc20_events` but use subqueries for the columns that are specific to the types of events you want.

Example trick: use a temp column (`event_data` or something) which gets populated with that subquery containing a comma separated list of values using `string_agg`. That allows you to get all required values and then just deserializing it when converting the row into a usable value in JS.